src/AppBundle/Service/ReportUtils.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace AppBundle\Service;
+
+class ReportUtils
+{
+    /**
+     * Generates and returns the report start date from a given end date.
+     * -365 days + 1 if note a leap day (otherwise we get 2nd March)
+     *
+     * @param \DateTime $reportEndDate
+     *
+     * @return \DateTime $reportStartDate
+     */
+    public static function generateReportStartDateFromEndDate(\DateTime $reportEndDate)
+    {
+        $reportStartDate = clone $reportEndDate;
+
+        $isLeapDay = $reportStartDate->format('d-M') == '29-Feb';
```

Comment:
i think you can use isValidDay on DateTime object for this

Reply:
either that or DateTime returns False if set to invalid day

Reply:
I've only moved the existing code from a non-static method in the class as it was unrelated the the class (and wasn't even static). I agree it should be refactored, doesn't look a good solution. I'll add a TODO comment to refactor later, branch big enough atm

Reply:
That method substract one year to the date, so the return value should always be valid. Not sure about the leapDay thing, needs more investigation and specs collect